src/workerd/api/tests/js-rpc-test.js

```diff
@@ -80,6 +99,26 @@ export class MyService extends WorkerEntrypoint {
   get nonFunctionProperty() { return {foo: 123}; }
 
   get functionProperty() { return (a, b) => a - b; }
+
+  get objectProperty() {
+    return {
+      func: (a, b) => a * b,
+      deeper: {
+        deepFunc: (a, b) => a / b,
+      },
+      counter5: new MyCounter(5),
+      nonRpc: new NonRpcClass(),
+      someText: "hello",
+    };
+  }
+
+  get promiseProperty() {
+    return scheduler.wait(10).then(() => 123);
+  }
+
+  get rejectingPromiseProperty() {
+    return Promise.reject(new Error("REJECTED"));
```

Comment:
I might just be overlooking it here, but is there also a test case with a synchronous throw?

Reply:
Also, a test verifying the behavior when a non-typical exception is thrown (e.g. an `AggegateError` triggered by `Promise.any([...])`. Our tunneled exception mechanism currently doesn't handle these directly so we likely should test the behavior.

Reply:
Added tests for throwing exceptions.

Not sure what you are asking for regarding non-typical exceptions. We know that these do not work, but it's because we're relying on the general exception tunneling behavior, which is tested elsewhere.